libcaf_core/caf/async/blocking_consumer.hpp

```diff
@@ -45,6 +45,7 @@ class blocking_consumer {
           return read_result::try_again_later;
       }
       auto [again, n] = buf_->pull_unsafe(guard, policy, 1u, *this);
+      guard.unlock();
```

Comment:
Can this function be called in parallel? Then, the check in line 36 could happen before the assignment in 49/50.

Reply:
This function itself cannot run in parallel. A `blocking_consumer` is for one caller/thread only.